src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java

```diff
@@ -9,11 +16,50 @@
 
 @Configuration
 @EnableWebSecurity
-public class WebSecurityConfig  extends WebSecurityConfigurerAdapter {
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private UserDetailsServiceImpl userDetailsService;
+
     @Override
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
-                .antMatchers("/", "/index").permitAll();
+                .antMatchers("/welcome").authenticated()
+                .and()
+                .csrf().disable()
+                .authorizeRequests()
+                .and()
+                .formLogin()//.loginPage("/signin").permitAll()
+                .defaultSuccessUrl("/welcome")
+                .usernameParameter("username")
+                .passwordParameter("password")
+                .failureUrl("/signin-error")
+                .and()
+                .logout().logoutUrl("/logout").logoutSuccessUrl("/")
+                .and()
+                .rememberMe().key("token").tokenValiditySeconds(3600);
+    }
+
+////We don't need a password encoder while we don't have registration
+//    @Bean
+//    public PasswordEncoder passwordEncoder() {
+//        return new BCryptPasswordEncoder();
+//    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth)
+            throws Exception {
```

Comment:
I am not sure whether I can do that, because as the documentation specifies, the method _authorizeRequests()_ from springframework security library throws _java.lang.Exception_. So when I try to replace Exception with the more specific AuthenticationException I get a message that  java.lang.Exception is unhandled. I tried to surround the chain of methods with try-catch but I think it won't make sense then, since my _configure(...)_ method consists of the chain of methods called one by one. Do you have any ideas on how I could fix it? Or in such situation when the Exception is thrown by the framework method, we should leave it as it is?Thank you!